### ISSUE TITLE
Week 5 proposed plan

### ISSUE THREAD

**Issue by rug1** [FAC6/FAC7#18]
# Week 5
## Monday

10-11:30 Morning Challenge

11:30-4:30 Readmes

Split into four jigsaw groups and each work on a different database:
- Redis
- MongoDB
- PostgreSQL
- LevelDB

Everyone should research these questions (30 minutes):
- What is a relational database?
- What is the difference between an SQL and NoSQL database?

For your specific database (1h30mins):
- How does the DB store information?
- What are the different data types that can be stored? e.g. strings, lists, hashes, sets
- What are the advantages and disadvantages of this database? e.g. speed, scalability, data structure, query performance

Write a guide on how to (1h30mins):
- Install the database
- Which npm module should you use for this database
- How to access the database from the command line

Stretch goal: present a working demo showing how to add to the database from a HTML form.

4:30-6 Presentations
## Tuesday

10-1 [Redis Workshop](https://github.com/msmichellegar/redis-node-workshop). Work in pairs. People who did the Redis readmes will have an advantage but if they finish early, they can help the others.

1-2 Lunch

2-5 [Websocket Tutorial](http://socket.io/get-started/chat/) Work in pairs. Tutorial uses express and jQuery, change it into a pure node server and vanilla javascript.

5-6 Questions on anything they want to go through or finish up either of the workshops if they haven't finished.
## Wednesday

10-11:30 Morning Challenge

11:30-11:45 Introduce project:
This week you will be creating a chat room that updates in real time (obviously). You will need to implement both socket.io and a database.
- Use socket.io for real time messages
- Use a database to store the messages
- Deploy your chatroom to heroku
##### Stretch Goals
- Make usernames
- The ability to create rooms
- See a list of users who are available online
##### Don't Forget
- NO FOR LOOPS ALLOWED
- Badges - code climate, coverage, code crystal, build
- Testing frontend and backend - at least 80% coverage on backend
- Continuous Integration - Set up travis
## Thursday

10-5:30 Carry on with projects

5:30-6 Adam Baldwin Node Security talk via Google hangouts
## Friday

10:30-12 Code Review

12-1 Respond to issues

1-2 Lunch/write up stop go continue

2-2:45 go through stop go continue

2:45-3:15 upstairs project

3:15-4 Business Development

4-6 Project Presentations

**Comment by rug1** [comment]
I'm not sure if the project might be a bit too challenging

**Comment by sofer** [comment]
@rug1 we might want to rename this to _Week 6 proposed plan_.

And following the suggestion from @iteles, we might even want to rename it to _Weeks 6-7 proposed plan_.

Also, we may want to keep a careful eye on what goes on this week, because we are taking a slightly different approach with READMEs (see https://github.com/FAC7/READMEs), which is going to involve people creating more structured tutorial READMEs and then actually getting the rest of the class to go through them. If that works, it may mean we will need more time (but we will have less work to do!).

**Comment by rug1** [comment]
Yes I was just about to suggest this could actually be the proposed plan for week 6
